### PR TITLE
feat: proper non-interactive-mechanic

### PR DESCRIPTION
**Pull request by justchokingaround**
# Pull Request Template

## Type of change

- [ ] Bug fix
- [ ] Feature
- [ ] Documentation update

## Description

*ramble here*

## Checklist

- [x] any anime playing
- [x] bumped version
---
- [ ] next, prev and replay work
- [ ] `-c` history and continue work
- [ ] `-d` downloads work
- [ ] `-s` syncplay works
- [ ] `-q` quality works
- [ ] `-v` vlc works
- [ ] `-e` select episode works
- [ ] `-S` select index works
- [ ] `-r` range selection works
- [ ] `--dub` and regular (sub) mode both work
- [ ] all providers return links (not necessarily on a single anime, use debug mode to confirm)
---
- [ ] `-h` help info is up to date
- [ ] Readme is up to date
- [ ] Man page is up to date

## Additional Testcases

- The safe bet: One Piece
- Episode 0: Saenai Heroine no Sodatekata ♭
- Unicode: Saenai Heroine no Sodatekata ♭
- Non-whole episodes: Tensei shitara slime datta ken (ep. 24.5, ep. 24.9)

### CODE REVIEW

**Comment by justchokingaround** [comment]
fuck version bump, i ain't no bitch

File: ani-cli [hunk 1]
```diff
@@ -189,7 +189,7 @@ get_episode_url() {
     done
     wait
     # select the link with matching quality
-    links=$(cat "$cache_dir"/* | sed 's|^Mp4-||g' | sort -g -r -s)
+    links=$(cat "$cache_dir"/* | sed 's|^Mp4-||g;/http/!d' | sort -g -r -s)
```
Comment (port19x) [inline ani-cli:192]: why the heck wouldn't we want encrypted traffic?

**Review by port19x — CHANGES_REQUESTED** [review]

File: ani-cli (hunk 1 above)
Comment (justchokingaround) [reply ani-cli:192]: `!` is the not operator mister

Comment (port19x) [reply ani-cli:192]: Oh, then PR title is off?
Forcing https is okay for me

Comment (justchokingaround) [reply ani-cli:192]: wym, it forces the video link to contain http

Comment (port19x) [reply ani-cli:192]: > why the heck wouldn't we want encrypted traffic?

hence my original concern

Comment (port19x) [reply ani-cli:192]: Or does it just sharpen the link matching to include either http or https, weeding out some strange false positive?

Comment (justchokingaround) [reply ani-cli:192]: it matches both http and https, basically any line that contains http in it

**Comment by port19x** [comment]
I haven't thoroughly tested my change yet, but should be good

File: ani-cli (hunk 1 above)
Comment (port19x) [reply ani-cli:192]: Fair, then it doesn't matter

**Review by port19x — APPROVED** [review]

**Comment by port19x** [comment]
I'd prefer another review before merge, as I added some lines too

**Comment by port19x** [comment]
Renamed to reflect the more impactful change

**Pull request merged**